src/algebra/algebra/basic.lean

```diff
@@ -728,6 +728,24 @@ by { ext, simp }
   alg_hom.comp ↑e.symm (e : A₁ →ₐ[R] A₂) = alg_hom.id R A₁ :=
 by { ext, simp }
 
+/-- An alg_equiv induces and equiv of alg_hom types -/
+def alg_hom_equiv_alg_hom_left (ϕ : A₁ ≃ₐ[R] A₂) : (A₁ →ₐ[R] A₃) ≃ (A₂ →ₐ[R] A₃) :=
+{ to_fun := λ f, f.comp ϕ.symm.to_alg_hom,
+  inv_fun := λ f, f.comp ϕ.to_alg_hom,
+  left_inv := λ f,
+    by { simp only [alg_hom.comp_assoc, to_alg_hom_eq_coe, symm_comp, alg_hom.comp_id] },
+  right_inv := λ f,
+    by { simp only [alg_hom.comp_assoc, to_alg_hom_eq_coe, comp_symm, alg_hom.comp_id] } }
+
+/-- An alg_equiv induces and equiv of alg_hom types -/
+def alg_hom_equiv_alg_hom_right (ϕ : A₁ ≃ₐ[R] A₂) : (A₃ →ₐ[R] A₁) ≃ (A₃ →ₐ[R] A₂) :=
+{ to_fun := λ f, ϕ.to_alg_hom.comp f,
+  inv_fun := λ f, ϕ.symm.to_alg_hom.comp f,
+  left_inv := λ f,
+    by { simp only [←alg_hom.comp_assoc, to_alg_hom_eq_coe, symm_comp, alg_hom.id_comp] },
+  right_inv := λ f,
+    by { simp only [←alg_hom.comp_assoc, to_alg_hom_eq_coe, comp_symm, alg_hom.id_comp] } }
```

Comment:
Over in `data/equiv/basic`, we don't create these separately, and call this `equiv.arrow_congr`. IMO, it would make sense to do the same thing here, and call this `alg_equiv.arrow_congr`.

Reply:
Ah, good idea. Will do.

Reply:
If you're feeling ambitious, `add_equiv`, `mul_equiv`, `ring_equiv`, and `linear_equiv` could all do with an `arrow_congr`.

Reply:
I'm not feeling that ambitious, but I've made the changes for alg_equiv